corehq/apps/export/forms.py

```diff
@@ -920,27 +922,37 @@ def get_filters(self, can_access_all_locations, accessible_location_ids, group_i
         :param date_range: A DatePeriod or DateSpan
         """
         form_filters = []
-        if can_access_all_locations:
-            form_filters += [_f for _f in [
-                self._get_group_filter(group_ids),
-                self._get_user_type_filter(user_types),
-            ] if _f]
 
-        form_filters += [_f for _f in [
-            self._get_users_filter(user_ids),
-            self._get_locations_filter(location_ids)
-        ] if _f]
+        if FILTER_ON_GROUPS_AND_LOCATIONS.enabled(self.domain_object.name):
+            if can_access_all_locations and user_types:
+                form_filters += self._get_user_type_filter(user_types)
+
+            form_filters += filter(None, [self._get_users_filter(user_ids)])
+
+        else:
+            if can_access_all_locations:
+                form_filters += filter(None, [self._get_group_filter(group_ids),
+                                              self._get_user_type_filter(user_types)])
+
+            form_filters += filter(None, [self._get_users_filter(user_ids),
+                                          self._get_locations_filter(location_ids)])
```

Comment:
It might be nicer to keep these ifs flatter

```
form_filters = [self._get_users_filter(user_ids)]
if user_types:
   form_filters += self._get_user_type_filter(user_types)
...
```